support/catalogs/images.go

```diff
@@ -101,8 +101,11 @@ func releaseVersionFn(ctx context.Context, hcp *hyperv1.HostedControlPlane, pull
 	}
 }
 
-func imageExistsFn(ctx context.Context, pullSecret []byte, imageMetadataProvider util.ImageMetadataProvider) func(image string) (bool, error) {
+func imageExistsFn(ctx context.Context, pullSecret []byte, imageMetadataProvider util.ImageMetadataProvider, guestCluster bool) func(image string) (bool, error) {
 	return func(image string) (bool, error) {
+		if guestCluster {
```

Comment:
Please add a unit test for this.

Reply:
I want to but I'm not sure if it's wise because the nature of the test. There currently is no testing for the `imageExistsFn` and I'm not sure if it'd work if I had to create one myself due to the test having to check if the image manifest actually exists.

Reply:
And I'm having trouble finding examples of testing a go variable function. As far as I can tell, there are none in the Hypershift repo other than `TestImageLookupCacheKeyFn` which only checks hash differences.

Reply:
There is a test of `imageLookupCacheKeyFn` so I suppose you could do a simple test that ensures `true, nil` is returned when `guestCluster` provided.